src/web/client/test/integration/portalSchemaReader.test.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import * as schemaHelperUtil from "../../utilities/schemaHelperUtil";
+import * as portalSchemaReadeMock from "../integration/portalSchemaReader.mock";
+import { schemaEntityKey } from "../../../client/schema/constants";
+import Sinon, { stub } from "sinon";
+import {
+    getEntitiesSchemaMap,
+    getDataSourcePropertiesMap,
+    getEntitiesFolderNameMap,
+} from "../../../client/schema/portalSchemaReader";
+import { expect } from "chai";
+
+describe("portalSchemaReader", () => {
+    afterEach(() => {
+        Sinon.restore();
+    });
+    it("getEntitiesSchemaMap", () => {
+        stub(schemaHelperUtil, "getPortalSchema").returns(
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            portalSchemaReadeMock.schema_data as any
+        );
+
+        const result = getEntitiesSchemaMap("testSchema");
+        for (const [key] of result) {
+            expect(key).equal("websites");
+        }
+        const websitesData = result.get("websites");
+        expect(websitesData?.get("_vscodeentityname")).equal("websites");
+        expect(websitesData?.get("_dataverseenityname")).equal(
+            "powerpagesites"
+        );
+        expect(websitesData?.get("_displayname")).equal("Website");
+    });
+
+    it("getEntitiesSchemaMap() - empty entity", () => {
+        stub(schemaHelperUtil, "getPortalSchema").returns(
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
1) Where is the assert if the schema itself is corrupted?
2) What happens if we send undefined/null schema to the function?
3) What happens if schema length is 0?
4) How you will handle if the return datatype is changed?
5) The schema is so exhaustive why so few metadata is present in your mocked schema?
6) Where you are testing for portal_schema_v1 and portal_Schema_V2
7) Your UT is not sufficient if the schema is changed. Your UT should verify each and every field of the schema

Reply:
Please Find the Below comments :

1. If the schema is corrupt the function it self throw an error
2. we will not able to send undefined or null because parameter is type of string.
3. Function will throw an exception
4. We don't have any assert to verify the type becuase typeof is also showing Map as 'object'
5. Implemented 
6. That will be test when we will be writing the test case for "schemaHelperUtil"
7. Implemented